internal/stage_6.go

```diff
@@ -37,7 +37,7 @@ func testHandlesMultipleConcurrentConnections(stageHarness *test_case_harness.Te
 		return err
 	}
 	logger.Debugf("Sending first set of requests")
-	for i := 0; i < connectionCount; i++ {
+	for i := connectionCount; i < 0; i-- {
```

Comment:
```suggestion
        // Test connections in reverse order so that we don't accidentally test the listen backlog
        // Ref: <PR link>
	for i := connectionCount; i < 0; i-- {
```

(This seems very non-obvious, so let's add a comment)